mne_bids/write.py

```diff
@@ -903,7 +905,7 @@ def make_dataset_description(path, name, data_license=None,
 
 def write_raw_bids(raw, bids_path, events_data=None,
                    event_id=None, anonymize=None,
-                   format='auto',
+                   format='auto', eeg_reference='n/a',
```

Comment:
I can live with it but not super enthusiastic

Instead I'd suggest having an API to find the 'n/a' entries in a dataset, return them as a dict and update them across the dataset. Didn't have an issue somewhere? We can't have a param for every entry in the json files ...

Reply:
```py
na_entries = find_entities(bids_root, cond='n/a')
na_entries.update(eeg_reference=eeg_reference)
save_entities(bids_root, na_entries)
```

Reply:
Could use this function: https://github.com/mne-tools/mne-bids/blob/35f3c3db02ab098b4fbe98c3add32b73a51ce768/mne_bids/sidecar_updates.py#L16

Reply:
nice @adam2392 you are the best!

Reply:
Yes, I'm aware of this function and it's already mentioned in the documentation.

My opinion is that for this specific use case – adding an EEG reference – users shouldn't have to do additional work after calling `write_raw_bids()` – partly because it's inconvenient, partly because some will simply forget about it (?)

Given there's such a plethora of potential parameters – and we cannot expect users to read the BIDS specification! – I'm wondering whether it could also make sense to have some kind of functionality that specifically helps with adding the REQUIRED and RECOMMENDED fields.

Look at the specs for `*_eeg.json`, for example:
https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/03-electroencephalography.html#sidecar-json-_eegjson

This list is sheer endless, and currently MNE-BIDS only fills out a few RECOMMENDED fields – and some of the REQUIRED fields are simply set to `'n/a'` to make the validator pass. That's not how things should be. And like I said, we shouldn't expect users to read the specs, esp. since they can change from revision to revision. We need to give users a hand.

I think this EEGLAB tool does a way better job in guiding users:
https://github.com/sccn/bids-matlab-tools/wiki#export-datasets-to-bids-from-the-graphic-interface

I wonder if we need something similar to make all those BIDS options really accessible to our users.

Reply:
okay fair enough. But is EEG reference special in any way from the other parameters? I totally agree that we need to hold the user's hand as much as we can. Would it make sense to:

1) Remove writing 'n/a' to the fields to hack the passing of validator?
2) Then add a function ... or maybe a new BIDS object (like the Info object in MNE) that prepopulates the fields? E.g., if you do:

```py
BIDSTemplate(data_type='eeg')
```

you will get a dict with empty fields. But if you do:

```py
BIDSTemplate(bids_root)
```

you will get a dict with fields populated with what is in the dataset. And then you can fix the examples so they explicitly update the required fields

Reply:
> you will get a dict with fields populated with what is in the dataset. And then you can fix the examples so they explicitly update the required fields

This is getting a little more complex if we consider that each participant and session can have different settings…

Reply:
Since `(i)EEGReference` is a REQUIRED field, then it makes sense to allow that to be passed to `write_raw_bids`, so I am more agreeing with @hoechenberger now... Are there many more REQUIRED fields that we aren't explicitly writing? 

### Other ideas
For more advanced usage, would we one day want to allow just ``**kwargs`` arguments to `write_raw_bids`, where a user can pass ANY REQUIRED fields to overwrite? 

Another idea that is probably unwieldy but will throw out there :p, could we create a dictionary template with the default values for all the REQUIRED and RECOMMENDED fields for MEG/EEG/iEEG and have the user import them, set certain fields themselves and then pass to `write_raw_bids`?